### DISCUSSION TITLE
multiplayer limits?

### DISCUSSION THREAD

**Discussion by dcsan** [boardgameio/boardgame.io#1015] [General]
hi! I'm trying to make a boardgame for a large number of users, I was wondering if there are limits to this system?
I assume its more designed for games of 4 or 8 players max, but if i wanted to use it more like a tilebased MMO RPG for 100s of players... wondering if there are examples running a large board with this engine?

**Comment by delucis** [comment]
There aren’t any framework-specific limits on player numbers. The limit you’d run into would presumably be server resources in terms of number of simultaneous socket connections for example. A single game shouldn’t require more server CPU for more players as the server will only process one action at a time per game. Another hypothetical bottleneck might be game state size if it is storing data for hundreds of players, but you would need to test if that is significant in your case.

boardgame.io is designed for turn-based play in general, rather than real-time games where many players may be taking simultaneous actions so that may be something else to consider. There is support for “simultaneous” actions using [stages](https://boardgame.io/documentation/#/stages) and the [`ignoreStaleStateID` long-form move option](https://boardgame.io/documentation/#/api/Game), but if that’s the main kind of play you want to support some other framework may be able to handle that more elegantly.

**Comment by dcsan** [comment]
thanks!
i'm trying to develop something like a large scale board / strategy game, not an action game. something like hero academy en masse - which is kind of like chess with RPG attributes. Since there are so many players it would be simultaneous resolution exactly!
